### PR TITLE
MISUV-8871 Migrate ObligationsConnector to HttpClientV2

### PR DESCRIPTION
**Pull request by dnyuar**

### CODE REVIEW

File: app/connectors/ObligationsConnector.scala [hunk 1]
```diff
@@ -20,16 +20,14 @@ import config.MicroserviceAppConfig
 import models.obligations.{ObligationsErrorModel, ObligationsModel, ObligationsResponseModel}
 import play.api.http.Status
 import play.api.http.Status._
-import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
+import uk.gov.hmrc.http.client.HttpClientV2
```
Comment (avelx) [inline app/connectors/ObligationsConnector.scala:23]: please Optimise imports

File: it/test/connectors/ObligationsConnectorISpec.scala [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import assets.BaseIntegrationTestConstants.testNino
+import assets.ReportDeadlinesIntegrationTestConstants._
+import helpers.{ComponentSpecBase, WiremockHelper}
```
Comment (avelx) [inline it/test/connectors/ObligationsConnectorISpec.scala:21]: Please optimise imports

File: app/connectors/ObligationsConnector.scala (hunk 1 above)
Comment (dnyuar) [reply app/connectors/ObligationsConnector.scala:22]: Updated

File: it/test/connectors/ObligationsConnectorISpec.scala (hunk 2 above)
Comment (dnyuar) [reply it/test/connectors/ObligationsConnectorISpec.scala:21]: Updated

File: app/connectors/ObligationsConnector.scala (hunk 1 above)
Comment (dnyuar) [reply app/connectors/ObligationsConnector.scala:22]: Updated

File: it/test/connectors/ObligationsConnectorISpec.scala (hunk 2 above)
Comment (dnyuar) [reply it/test/connectors/ObligationsConnectorISpec.scala:21]: Updated

**Review by avelx — APPROVED** [review]

**Pull request merged**